### ISSUE TITLE
How do I fix the order of the logs?

### ISSUE THREAD

**Issue by AlessioGr** [Julien-R44/pino-loki#17]
Hi there,

the order of the logs in Loki is incorrect. If I send two consecutive logs, they may or may not appear in the wrong order.

E.g. these two:
![Screenshot 2023-05-21 at 10 16 29@2x](https://github.com/Julien-R44/pino-loki/assets/70709113/61b138aa-aac1-4537-a56a-7348b92afe87)
are in the wrong order.

Any idea how that could be fixed? Maybe the timestamp isn't accurate enough?

**Comment by Julien-R44** [comment]
Are you using `batching` with `replaceTimestamps`? 

If not, maybe using more precise timestamps can solve the problem, yes. I'll try to look at that for the next version

**Comment by AlessioGr** [comment]
> Are you using `batching` with `replaceTimestamps`?
> 
> If not, maybe using more precise timestamps can solve the problem, yes. I'll try to look at that for the next version

Nope! Currently using neither. Can I use `replaceTimestamps` to replace it with more precise timestamps?

**Comment by Julien-R44** [comment]
Nope, `replaceTimestamps` will just replace timestamps provided by pino

I just added a support for nano timestamps in v2.1.0, see example here : 
https://github.com/Julien-R44/pino-loki/blob/main/examples/custom_timestamp.ts

We are using pino API with the `timestamp` function : 
https://github.com/pinojs/pino/blob/master/docs/api.md#timestamp-boolean--function

Let me know if it works for you, otherwise feel free to re-open the issue

( Btw thanks for the tip ! 😁 )

**Comment by AlessioGr** [comment]
Oh wow, thank you a TON, that was quick!

I'm currently testing it out, but getting that lovely error:
/Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/pino-loki/dist/index.cjs:5
const got = require('got');
            ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/got/dist/source/index.js from /Users/alessio/Documents/GitHub/payload-plugin-ai-impl/node_modules/pino-loki/dist/index.cjs not supported.

You really have to love JavaScript 😅 Did you change anything regarding esm/cjs with the update by any chance?

**Comment by Julien-R44** [comment]
Oh shit, yes the latest version of Got is ESM only. Since I migrated pino-loki to ESM ( but we're still dual-packaging ESM/CJS, no worries ), I hadn't noticed it. 

I'm going to downgrade this dependency and make a new release tomorrow morning. Sorry about that

**Comment by AlessioGr** [comment]
> Oh shit, yes the latest version of Got is ESM only. Since I migrated pino-loki to ESM ( but we're still dual-packaging ESM/CJS, no worries ), I hadn't noticed it.
> 
> I'm going to downgrade this dependency and make a new release tomorrow morning. Sorry about that

Ahh I see all good! Thank you a lot!! ❤️

hope I can make the switch to pure ESM soon and abandon the CJS hellhole haha

**Comment by Julien-R44** [comment]
Fixed with 2.1.3 !

**Comment by AlessioGr** [comment]
> Fixed with 2.1.3 !

Thank you a lot, working perfectly now! Also, in case anyone is wondering, this is how I made it to work with pino-pretty (by default it does not format the nano time):
```js
// pino-pretty-transport.js
function nanoToHumanReadable(nanoTimestamp) {
    // Check if the timestamp is a number or a BigInt
    let milliTimestamp;
    if (typeof nanoTimestamp === 'bigint') {
      milliTimestamp = Number(nanoTimestamp / BigInt(1e6));
    } else if (typeof nanoTimestamp === 'number') {
      milliTimestamp = Math.floor(nanoTimestamp / 1e6);
    } else {
      throw new Error('Timestamp must be a number or a BigInt');
    }
  
    // Create new Date object using the millisecond timestamp
    const date = new Date(milliTimestamp);
  
    // Retrieve hours, minutes, and seconds
    const hours = date.getHours();
    const minutes = date.getMinutes();
    const seconds = date.getSeconds();
  
    // Convert to two digits
    const twoDigitHours = String(hours).padStart(2, '0');
    const twoDigitMinutes = String(minutes).padStart(2, '0');
    const twoDigitSeconds = String(seconds).padStart(2, '0');
  
    // Return as formatted string
    return `${twoDigitHours}:${twoDigitMinutes}:${twoDigitSeconds}`;
  }

module.exports = opts => require('pino-pretty')({
    ...opts,
    //messageFormat: (log, messageKey) => `hello ${log[messageKey]}`,
    customPrettifiers: {
        // The argument for this function will be the same
        // string that's at the start of the log-line by default:
        time: timestamp => `🕰 ${nanoToHumanReadable(timestamp)}`,
      }
  })
```

Then my logger options:
```ts
const loadNs = process.hrtime();
const loadMs = new Date().getTime();
function nanoseconds() {
  let diffNs = process.hrtime(loadNs);
  return BigInt(loadMs) * BigInt(1e6) + BigInt(diffNs[0] * 1e9 + diffNs[1]);
}

export const loggerOptions: LoggerOptions = {
  //timestamp: () => `,"time":${Date.now()},"uniqueID":"${process.hrtime.bigint()}"`,
  timestamp: () => `,"time":${nanoseconds()}`,
  transport: {
    targets: [
      {
        //target: 'pino-pretty',
        target: './pino-pretty-transport.js',
        options: {
          colorize: true,
          ignore: 'pid,hostname',
          translateTime: 'HH:MM:ss',
        },
        level: null,
      },
      ...
```
